### PR TITLE
2680 fix option set import/export i18n

### PR DESCRIPTION
**Pull request by lmatiolis**
Fix option set import preferred_locale and also export.
Import wasn't using the ones from configatron and export was just using the name for the option, not one
of it's translations. Now we are getting the translation for the first language preferred.

### CODE REVIEW

File: app/models/option_node.rb [hunk 1]
```diff
@@ -232,7 +232,7 @@ def arrange_as_rows(hash = nil, parent_path = [])
       # output a row if we've hit a leaf node or a parent node with coordinates
       if children.empty? || (option_set.allow_coordinates? && node.option.has_coordinates?)
         # use the option path to construct the list of cell values
-        row = [node.id, *path.map(&:name)]
+        row = [node.id, *preferred_name_translations(path)]
```
Comment (hooverlunch) [inline app/models/option_node.rb]: Need to also fix the comment on line 225 to say 'mission preferred locale'.

File: app/models/option_node.rb [hunk 2]
```diff
@@ -245,6 +245,12 @@ def arrange_as_rows(hash = nil, parent_path = [])
     end
   end
 
+  def preferred_name_translations(path)
+    path.map do |path|
+      path.name_translations[configatron.preferred_locales.first.to_s]
```
Comment (hooverlunch) [inline app/models/option_node.rb]: We should really add a `mission_preferred_locale` entry to configatron. Would be much cleaner and reusable.

File: app/models/option_node.rb [hunk 3]
```diff
@@ -245,6 +245,12 @@ def arrange_as_rows(hash = nil, parent_path = [])
     end
   end
 
+  def preferred_name_translations(path)
+    path.map do |path|
```
Comment (hooverlunch) [inline app/models/option_node.rb]: Using `path` as loop var is confusing. For short loops I usually use the initials of the obj class. So in this case you could just use `o`.

File: app/models/option_set_import.rb [hunk 4]
```diff
@@ -199,4 +202,9 @@ def blank_interior_cell?(row)
       !row[row.index(nil)..-1].all?(&:nil?)
     end
 
+    def name_locale_key
+      preferred_locale = configatron.preferred_locales.first
+      name_locale_key = "name_#{preferred_locale}".to_sym
```
Comment (hooverlunch) [inline app/models/option_set_import.rb]: If we use the new configatron key you can combine these two lines and remove the assignment.

**Comment by hooverlunch** [comment]
Just a few comments and looks good to merge.

**Comment by lmatiolis** [comment]
@hooverlunch Please check if i did set the preferred_locale in the right way on the configatron.

File: app/models/option_set_import.rb [hunk 5]
```diff
@@ -199,4 +202,7 @@ def blank_interior_cell?(row)
       !row[row.index(nil)..-1].all?(&:nil?)
     end
 
+    def name_locale_key
+      "name_#{configatron.preferred_locale}".to_sym
```
Comment (hooverlunch) [inline app/models/option_set_import.rb]: Yes!

Comment (lmatiolis) [inline app/models/option_set_import.rb]: Bonus points earned! :)

**Pull request merged**